redbot/cogs/warnings/warnings.py

```diff
@@ -422,6 +428,29 @@ def __init__(self, bot: Red):
         if user_id == ctx.author.id:
             return await ctx.send(_("You cannot remove warnings from yourself."))
 
+        custom_allowed = await self.config.guild(ctx.guild).allow_custom_reasons()
+        guild_settings = self.config.guild(ctx.guild)
+        reason_type = None
+        async with guild_settings.reasons() as registered_reasons:
+            if reason.lower() not in registered_reasons:
+                msg = _("That is not a registered reason.")
+                if custom_allowed:
+                    reason_type = {"description": reason}
+                elif (
+                    ctx.guild.owner == ctx.author
+                    or ctx.channel.permissions_for(ctx.author).administrator
+                    or await ctx.bot.is_owner(ctx.author)
+                ):
+                    msg += " " + _(
+                        "Run `{prefix}warningset allowcustomreasons true` to enable custom "
+                        "reasons."
+                    ).format(prefix=ctx.prefix)
+                    return await ctx.send(msg)
+            else:
+                reason_type = registered_reasons[reason.lower()]
+        if reason_type is None:
+            return
```

Comment:
I still think we should return a user friendly message when a valid reason is not present, just silently doing nothing is very confusing imo

Reply:
If another QA/Core-dev disagrees let me know and i'll review this as it is.

Reply:
I do agree we shouldn't fail silently, though I don't think we should list all registered reasons, especially not in scope of this PR.